clientdb/order.go

```diff
@@ -728,10 +752,55 @@ func serializeOrderTlvData(w io.Writer, o order.Order) error {
 		orderChannelType, &channelType,
 	))
 
+	if len(o.Details().AllowedNodeIDs) > 0 {
+		bytes := FlattenPubKeySlice(o.Details().AllowedNodeIDs)
+		tlvRecords = append(
+			tlvRecords, tlv.MakePrimitiveRecord(
+				allowedNodeIDsType, &bytes,
+			),
+		)
+	}
+
+	if len(o.Details().NotAllowedNodeIDs) > 0 {
+		bytes := FlattenPubKeySlice(o.Details().NotAllowedNodeIDs)
+		tlvRecords = append(
+			tlvRecords, tlv.MakePrimitiveRecord(
+				notAllowedNodeIDsType, &bytes,
+			),
+		)
+	}
+
 	tlvStream, err := tlv.NewStream(tlvRecords...)
 	if err != nil {
 		return err
 	}
 
 	return tlvStream.Encode(w)
 }
+
+// FlattenPubKeySlice returns a flatten array of bytes from the
+// given array of public keys.
+func FlattenPubKeySlice(pubKeys [][33]byte) []byte {
+	flatten := make([]byte, len(pubKeys)*33)
+
+	for i, key := range pubKeys {
+		copy(flatten[i*33:(i+1)*33], key[:])
+	}
+
+	return flatten
+}
+
+// AssemblePubKeySlice is the inverse of FlattenPubKeySlice. Given a slice of
+// bytes representing a flattened slice of pubkeys returns the original slice
+// of pubkeys.
+func AssemblePubKeySlice(flatten []byte) [][33]byte {
+	numOfPubKeys := len(flatten) / 33
```

Comment:
nit: we could add a check make sure that `len(flatten) % 33 == 0`

Reply:
+1 to this sanity check